content/en/BestPractices/prevent-security-misconfig.md

```diff
@@ -0,0 +1,160 @@
+
+Security misconfigurations are failures to implement the required security controls in a software component. Misconfigurations may lead to:
+
+- Disclosure of sensitive information
+- Data or system compromise
+
+## Examples
+
+Let's explore some common examples of software misconfiguration.
+
+### Debug Mode Enabled
+
+A developer may leave the debug mode enabled in the production environment. Debug information may contain sensitive data such as:
+
+- Application internal paths
+- Secret keys
+- Code snippets
+
+To prevent this:
+
+- Configure security settings for your language and framework
+- Disable the debug mode in the production environment
+     - For a .NET application, update the `Web.config` or `Machine.config` file:
+
+          ```
+          <compilation debug="false"/>
+          ```
+
+### Verbose Error Messages
+
+Similar to the [previous example](#debug-mode-enabled), an attacker may retrieve sensitive data from verbose error messages or a stack trace.
+
+To prevent this:
+
+- Don't expose a stack trace to users
+- Configure custom error messages for your software
+     - For a .NET application, modify the `Web.config` file:
+
+          ```
+          <system.web>
+          <customErrors mode="RemoteOnly" defaultRedirect="Error.aspx">
+          <error statusCode="401" redirect="CustomPage.aspx" />
+          <error statusCode="403" redirect="CustomPage.aspx" />
+          <error statusCode="404" redirect="NotFound.aspx" />
+          <error statusCode="500" redirect="Error.aspx" />
+          </customErrors>
+          </system.web>
+          ```
+
+### Directory Listing
+
+Directory listing allows users to see all files and directories stored on a web server. A malicious actor may get unauthorized access to sensitive files. Search the web with `intitle: "index of /"` to see websites that have directory listing enabled.
+
+In the following example, we show how to prevent this problem for an Apache Tomcat web server:
+
+1. Open the `conf/web.xml` file in the directory where Tomcat is installed.
+
+     - In Windows 10, the default installation directory is `C:\Program Files (x86)\Apache Software Foundation\Tomcat 9.0`.
+
+1. In the file, set `<param-value>false</param-value>` to `false` for `<param-name>listings</param-name>`:
+
+     ```
+     <servlet>
+          <servlet-name>default</servlet-name>
+          <servlet-class>org.apache.catalina.servlets.DefaultServlet</servlet-class>
+          <init-param>
+               <param-name>debug</param-name>
+               <param-value>0</param-value>
+          </init-param>
+          <init-param>
+               <param-name>listings</param-name>
+               <param-value>false</param-value>
+          </init-param>
+          <load-on-startup>1</load-on-startup>
+     </servlet>
+     ```
+
+1. In the Apache web server, create a `.htaccess` file in the related directory.
+
+1. Add the following lines to the `httpd.conf` file, or replace existing lines with:
```

Comment:
The actual configuration file name varies. IIRC, while it's `httpd.conf` for Red Hat Enterprise Linux, it's `apache2.conf` on at least Ubuntu (and I think Debian-based systems). And in some cases, Apache is set up to take custom config files in a `conf.d/` subdirectory. So I suggest:

```suggestion
1. Add the following lines to the Apache web server configuration file file, or replace existing lines with:
```